src/client/app/effects/github.ts

```diff
@@ -111,7 +112,13 @@ export class GitHubEffects {
             return this._github.createOrUpdateGist(
                 description, files, gistId, type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
             )
-            .map((gist: IGist) => ({ gist: gist, snippetId: id }));
+            .map((gist: IGist) => ({ gist: gist, snippetId: id }))
+            .catch(exception => {
+                if (!gistOwnerId && exception.status >= 400 && exception.status <= 499) {
+                    throw new Error(JSON.stringify({ id: id, gistOwnerId: UNKNOWN_GIST_OWNER_ID }));
```

Comment:
I believe if key and value are the same in the object you can just use { id, gistOwnerId: UNKNOWN_GIST_OWNER_ID }